src/cli/commands/config/edit.js

```diff
@@ -26,11 +30,91 @@ module.exports = Command.extend({
   },
 
   run: (name) => {
-    var node = new IPFS()
-    node.version((err, version) => {
-      if (err) { return log.error(err) }
+    const node = new IPFS()
+    const editor = process.env.EDITOR || 'vi'
 
-      console.log(version)
+    function getConfig (next) {
+      node.config.show((err, config) => {
+        if (err) {
+          log.error(err)
+          next(new Error('failed to get the config'))
+        }
+
+        next(null, config)
+      })
+    }
+
+    function saveTempConfig (config, next) {
+      temp.open('ipfs-config', (err, info) => {
+        if (err) {
+          log.error(err)
+          next(new Error('failed to open the config'))
+        }
+
+        fs.write(info.fd, JSON.stringify(config, null, 2))
+        fs.close(info.fd, (err) => {
+          if (err) {
+            log.error(err)
+            next(new Error('failed to open the config'))
+          }
+        })
+
+        next(null, info.path)
+      })
+    }
+
+    function openEditor (path, next) {
+      const child = spawn(editor, [path], {
+        stdio: 'inherit'
+      })
+
+      child.on('exit', (err, code) => {
+        if (err) {
+          log.error(err)
+          throw new Error('error on the editor')
+        }
+
+        next(null, path)
+      })
+    }
+
+    function readTempConfig (path, next) {
+      fs.readFile(path, 'utf8', (err, data) => {
+        if (err) {
+          log.error(err)
+          next(new Error('failed to get the updated config'))
+        }
+
+        try {
+          next(null, JSON.parse(data))
+        } catch (err) {
+          log.error(err)
+          next(new Error(`failed to parse the updated config "${err.message}"`))
+        }
+      })
+    }
+
+    function saveConfig (config, next) {
+      node.config.replace(config, (err) => {
+        if (err) {
+          log.error(err)
+          next(new Error('failed to save the config'))
+        }
+
+        next()
+      })
+    }
+
+    async.waterfall([
+      getConfig,
+      saveTempConfig,
+      openEditor,
+      readTempConfig,
+      saveConfig
+    ], (err) => {
+      if (err) {
+        throw err
```

Comment:
`ronin` catches it and prints it properly:
<img width="401" alt="screen shot 2016-03-04 at 18 39 31" src="https://cloud.githubusercontent.com/assets/1211330/13536562/856ef294-e238-11e5-8928-0acf6bacfba5.png">